### PR TITLE
[DM-786] Create Resubscribe UI

### PR DESCRIPTION
**Pull request by VicThorX**
Hi team!

This a simple implementation inside of the `Domain Configuration` area, of the resubscribe functionality.

### UI view:

image

image

### Success view:

image

### Error view:

image

image

### Database:

image

I still need to do:
- [x] Add a message or action when the request is successful.

### CODE REVIEW

File: src/wwwroot/services/settings.js [hunk 1]
```diff
@@ -231,5 +232,25 @@
         url: RELAY_CONFIG.cuitServiceBaseUrl + '/taxinfo/by-cuit/' + cuit
       });
     }
+
+    function resubscribeEmailAddress (subscriberEmail, reason, domainName, onExpectedError) {
```
Comment (darianbenito) [inline src/wwwroot/services/settings.js]: A little thing, space between function name and params.

Comment (andresmoschini) [reply src/wwwroot/services/settings.js]:
```suggestion
    function resubscribeEmailAddress(subscriberEmail, reason, domainName, onExpectedError) {
```

Good catch @ms-darianbenito! you can make it easy to understand using that "suggestion" GitHub's feature:

image

File: src/wwwroot/services/settings.js [hunk 2]
```diff
@@ -231,5 +232,25 @@
...
+        + subscriberEmail
+        + '/resubscribe';
+
+      var data = {};
+      data['reason'] = reason;
+      data['domain'] = domainName;
+
+      return $http({
+        actionDescription: 'action_requesting_resubscribe_email_address',
+        //tryHandleError: function(rejection){ return tryHandleErrorDomainManager(rejection, onExpectedError); },
```
Comment (andresmoschini) [inline src/wwwroot/services/settings.js]: What is this commented code?

File: src/wwwroot/services/settings.js [hunk 3]
```diff
@@ -231,5 +232,25 @@
...
+    function resubscribeEmailAddress (subscriberEmail, reason, domainName, onExpectedError) {
+      var url = RELAY_CONFIG.baseUrl
+        + '/accounts/' + auth.getAccountName()
+        + '/subscribers/'
+        + subscriberEmail
+        + '/resubscribe';
+
+      var data = {};
+      data['reason'] = reason;
+      data['domain'] = domainName;
```
Comment (andresmoschini) [inline src/wwwroot/services/settings.js]:
```suggestion
      var data = {
        reason: reason,
        domain: domainName
      };
```

**Comment by andresmoschini** [comment]
Great work!

Before merging it you should:

* Add unit tests
* Deal with known errors

File: src/wwwroot/controllers/dKimConfigCtrl.js [hunk 4]
```diff
@@ -44,5 +46,15 @@
...
   }
+
+  vm.sendResubscribe = function(form) {
+    vm.submitted = true;
+
+    if (!form.$valid) {
+      return;
+    }
+
+    settings.resubscribeEmailAddress(vm.email, vm.reason, vm.domain, false);
```
Comment (andresmoschini) [inline src/wwwroot/controllers/dKimConfigCtrl.js]: I think that the 4th parameter is not necessary (at least by the moment)

```suggestion
    settings.resubscribeEmailAddress(vm.email, vm.reason, vm.domain);
```

File: src/wwwroot/services/settings.js [hunk 5]
```diff
@@ -231,5 +232,25 @@
...
+
+      return $http({
+        actionDescription: 'action_requesting_resubscribe_email_address',
+        //tryHandleError: function(rejection){ return tryHandleErrorDomainManager(rejection, onExpectedError); },
+        method: 'POST',
+        data: data,
```
Comment (darianbenito) [inline src/wwwroot/services/settings.js]:
```suggestion
        data: {
        reason: reason,
        domain: domainName
      },
```
just to continue the convention of most other methods in the file

File: src/wwwroot/locales/es-translation.js [hunk 6]
```diff
@@ -273,6 +276,8 @@ window['relay-translation-es'] = {
   "dkim_configuration_help_spf_subtitle": "Incluye el SPF de {{ companyName }} para evitar la falsificación de direcciones de tu remitente. A continuación puedes encontrar un ejemplo de cómo debería quedar el registro para utilizar el SPF de {{ companyName }}. Es recomendable que el registro sea de tipo TXT y no SPF.",
   "dkim_configuration_help_spf_subtitle_link": "Aquí puedes encontrar ayuda para configurar el SPF.",
   "dkim_configuration_help_spf_subtitle_link_href": "es/como-configurar-tu-dkim-paso-a-paso#spf",
+  "resubscribe_title": "Resubscribir dirección de correo",
+  "resubscribe_help_field": "Aquí puede eliminar una dirección de correo electrónico del estado 'Dropped'",
```
Comment (RodrigoPereyraDiaz) [inline src/wwwroot/locales/es-translation.js]: What do you think about this change to be equivalent to English text?
```suggestion
  "resubscribe_help_field": "Aquí puede remover una dirección de correo electrónico del estado 'Dropped'",
```

File: src/wwwroot/services/settings.js (hunk 2 above)
Comment (VicThorX) [reply src/wwwroot/services/settings.js]: I forgot to delete it, it is to handle errors, but it is for when we register new domains.

Comment (andresmoschini) [reply src/wwwroot/services/settings.js]: What do you think about this (for the moment):

```suggestion
        // TODO: add tryHandleError to deal with expected errors
```

File: src/wwwroot/controllers/ResubscribeCtrl.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    if (rejectionData.status == 400 && rejectionData.errorCode == 13) {
+      errorMessage = 'resubscribe_domain_error';
+    }
+
+    if (rejectionData.status == 400 && rejectionData.errorCode == 14) {
+      errorMessage = 'resubscribe_unsubscription_error';
+    }
+
+    if (rejectionData.status == 429 && rejectionData.errorCode == 1) {
+      errorMessage = 'resubscribe_too_many_request_error';
```
Comment (andresmoschini) [inline src/wwwroot/controllers/ResubscribeCtrl.js:66]: Nice!

**Comment by darianbenito** [comment]
✔️

**Pull request merged**